libraries/canjs/src/components.js

```diff
@@ -143,6 +143,8 @@ Component.extend({
     </div>
   `),
   ViewModel: DefineMap.extend({
+    seal: false// this allows webcomponentsjs to define $ on the viewModel
```

Comment:
I’m not really happy with this line. By default, instances of `can-define/map/map` are [sealed](https://canjs.com/doc/can-define/map/map.seal.html), and I get the following error without setting `seal` to `false`:

```
	TypeError: can't define property "$": Object is not extensible (node_modules/@webcomponents/webcomponentsjs/webcomponents-lite.js:71)
	dispatchEvent@node_modules/@webcomponents/webcomponentsjs/webcomponents-lite.js:142:342
	dispatch@webpack:///~/can-util/dom/events/events.js:68:0 <- tests.webpack.js:2086:9
	module.exports@webpack:///~/can-util/dom/dispatch/dispatch.js:18:0 <- tests.webpack.js:3957:9
	handler@webpack:///~/can-util/dom/events/make-mutation-event/make-mutation-event.js:47:0 <- tests.webpack.js:14135:8
	dispatchIfListening/<@webpack:///~/can-util/dom/mutation-observer/document/document.js:25:0 <- tests.webpack.js:14208:3
	dispatchIfListening@webpack:///~/can-util/dom/mutation-observer/document/document.js:24:0 <- tests.webpack.js:14207:2
	setup/mutationData.handler/</<@webpack:///~/can-util/dom/mutation-observer/document/document.js:107:0 <- tests.webpack.js:14290:9
	each@webpack:///~/can-util/js/each/each.js:19:0 <- tests.webpack.js:231:9
	setup/mutationData.handler/<@webpack:///~/can-util/dom/mutation-observer/document/document.js:100:0 <- tests.webpack.js:14283:6
	setup/mutationData.handler@webpack:///~/can-util/dom/mutation-observer/document/document.js:99:0 <- tests.webpack.js:14282:5
	add/observer</<@webpack:///~/can-util/dom/mutation-observer/document/document.js:41:0 <- tests.webpack.js:14224:7
	add/observer<@webpack:///~/can-util/dom/mutation-observer/document/document.js:40:0 <- tests.webpack.js:14223:6
```

I’m having a hard time tracking down what’s causing that error because it doesn’t happen when I debug it in the browser and `webcomponents-lite.js` is minified; looking through the un-minified source, I can see a couple places where `$` is being set, but I’m not sure why.

Reply:
@azakus do you know if/where wc-lite defines `$`?

Reply:
GAH, this looks like a closure compiler issue. The compiled version of this code https://github.com/webcomponents/shadydom/blob/master/src/patch-events.js#L311 turns the `__eventWrappers` property of`fnOrObject`, the thing you pass as the second argument to `addEventListener`, into `$`.

I'll make a PR for ShadyDOM, and then a new version for webcomponentsjs.

Reply:
https://github.com/webcomponents/shadydom/pull/174

Reply:
Ok, should be fixed with webcomponentsjs 1.0.10

Reply:
Thank you @azakus! I will test this and update my PR after https://github.com/webcomponents/custom-elements-everywhere/pull/11 gets merged in. 😊

Reply:
@azakus After looking at the `shadydom` change more thoroughly, I understand that it just changed the name of the property that’s set.

The problem I meant to describe in this thread is that _any_ property is set when I call `addEventListener` on a custom element; by default, my `viewModel` object is [sealed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal), so it would be nice if `shadydom` didn't mutate it.

I can work around it just fine, but I thought it was worth bringing up. Thanks for your help! 😊